src/map_field.cpp

```diff
@@ -248,12 +248,15 @@ void map::spread_gas( field_entry &cur, const tripoint &p, int percent_spread,
     const int winddirection = g->weather.winddirection;
     const int windpower = get_local_windpower( g->weather.windspeed, cur_om_ter, p, winddirection,
                           sheltered );
-    // Reset nearby scents to zero
+
+    const int current_intensity = cur.get_field_intensity();
+    const field_type_id ft_id = cur.get_field_type();
+
+    // modify scents by neutralization value (minus)
     for( const tripoint &tmp : points_in_radius( p, 1 ) ) {
-        sblk.apply_gas( tmp );
+        sblk.apply_gas( tmp, ft_id->get_intensity_level( current_intensity ).scent_neutralization );
```

Comment:
Two minor things.
Capture the value of scent_neutralization outside the loop.
Skip the loop entirely of the value of scent_neutralization is 0.